Sources/XcodeProj/Objects/Files/PBXFileSystemSynchronizedRootGroup.swift

```diff
@@ -0,0 +1,108 @@
+    ///   - tabWidth: the visual width of tab characters
+    ///   - wrapsLines: should the IDE wrap lines when editing the object?
+    ///   - explicitFileTypes: It maps relative paths inside the synchronized root group to a particular file type.
+    ///   - exceptions: It returns a list of exception objects that override the configuration for some children in the synchronized root group.
+    ///   - explicitFolders: A list of relative paths to children folder whose configuration is overriden.
+    public init(sourceTree: PBXSourceTree? = nil,
+                path: String? = nil,
+                name: String? = nil,
+                includeInIndex: Bool? = nil,
+                usesTabs: Bool? = nil,
+                indentWidth: UInt? = nil,
+                tabWidth: UInt? = nil,
+                wrapsLines: Bool? = nil,
+                explicitFileTypes: [String: String] = [:],
+                exceptions: [PBXFileSystemSynchronizedBuildFileExceptionSet] = [],
+                explicitFolders: [String] = []) {
+        self.explicitFileTypes = explicitFileTypes
+        self.exceptionsReferences = exceptions.references()
+        self.explicitFolders = explicitFolders
+        super.init(sourceTree: sourceTree,
+                   path: path,
+                   name: name,
+                   includeInIndex: includeInIndex,
+                   usesTabs: usesTabs,
+                   indentWidth: indentWidth,
+                   tabWidth: tabWidth,
+                   wrapsLines: wrapsLines)
+    }
+    
+    // MARK: - Decodable
+    
+    fileprivate enum CodingKeys: String, CodingKey {
+        case explicitFileTypes
+        case exceptions
+        case explicitFolders
+    }
+    
+    public required init(from decoder: Decoder) throws {
+        let objects = decoder.context.objects
+        let objectReferenceRepository = decoder.context.objectReferenceRepository
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.explicitFileTypes = (try container.decodeIfPresent(.explicitFileTypes)) ?? [:]
+        let exceptionsReferences: [String] = (try container.decodeIfPresent(.exceptions)) ?? []
+        self.exceptionsReferences = exceptionsReferences.map { objectReferenceRepository.getOrCreate(reference: $0, objects: objects) }
+        self.explicitFolders = (try container.decodeIfPresent(.explicitFolders)) ?? []
+        try super.init(from: decoder)
+    }
+    
+    // MARK: - PlistSerializable
+    
+    override var multiline: Bool { false }
+
+    override func plistKeyAndValue(proj: PBXProj, reference: String) throws -> (key: CommentedString, value: PlistValue) {
+        var dictionary: [CommentedString: PlistValue] = try super.plistKeyAndValue(proj: proj, reference: reference).value.dictionary ?? [:]
+        dictionary["isa"] = .string(CommentedString(type(of: self).isa))
+        dictionary["exceptions"] = .array(self.exceptionsReferences.map({ exceptionReference in
```

Comment:
It seems as of Xcode 16 beta 4 the `exceptions` attribute is omitted when empty as opposed to being set as empty. Performing a read / write test I see the following diff (reading from a newly created project in Xcode 16 UI).


```diff
-               FB50EE602C5A922900C43B53 /* FolderRefs */ = {isa = PBXFileSystemSynchronizedRootGroup; explicitFileTypes = {}; explicitFolders = (); path = FolderRefs; sourceTree = "<group>"; };
+               FB50EE602C5A922900C43B53 /* FolderRefs */ = {isa = PBXFileSystemSynchronizedRootGroup; exceptions = (); explicitFileTypes = {}; explicitFolders = (); path = FolderRefs; sourceTree = "<group>"; };
```